src/main/java/io/jenkins/plugins/gitlabserverconfig/servers/GitLabServer.java

```diff
@@ -278,6 +278,13 @@ public Secret getSecretToken() {
         return secretToken;
     }
 
+    public String getSecretTokenAsPlainText() {
+        if (this.secretToken == null) {
+            return null;
```

Comment:
>        if (this.secretToken == null) {
>            return null;
Shouldn't this be returning empty string rather than **null**? 
Otherwise [this](https://github.com/LinuxSuRen/gitlab-branch-source-plugin/blob/55f62dd1747f6badc42a963dd4684cfe1d4fbdd3/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabWebHookAction.java#L89) 
> if(server.getSecretTokenAsPlainText().equals(secretToken) || (server.getSecretTokenAsPlainText().isEmpty() && secretToken == null)) {

and [this](https://github.com/LinuxSuRen/gitlab-branch-source-plugin/blob/55f62dd1747f6badc42a963dd4684cfe1d4fbdd3/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSystemHookAction.java#L89) 
> if(server.getSecretTokenAsPlainText().equals(secretToken)) {

will still NPE.

Reply:
I can verify that we still got the NPE